### PR TITLE
Add global ocean test case with Lagrangian particles

### PR DESCRIPTION
**Pull request by mark-petersen**
Add a test that creates an initial condition particle file, then runs LIGHT for the forward global ocean run.

### CODE REVIEW

File: compass/ocean/tests/global_ocean/init/initial_state.py [hunk 1]
```diff
@@ -163,5 +164,9 @@ def run(self):
         add_mesh_and_init_metadata(self.outputs, config,
                                    init_filename='initial_state.nc')
 
+        particles.write(init_filename='initial_state.nc', particle_filename='particles.nc',
+                        graph_filename='graph.info.part.{}'.format(cores),
+                        types='buoyancy')
```
Comment (mark-petersen) [inline compass/ocean/tests/global_ocean/init/initial_state.py:169]: I copied this particle creation line from the ziso test. I wasn't sure where to put it - here (would run every time) or within the `lagrangian_particles` test.

File: compass/ocean/tests/global_ocean/lagrangian_particles/__init__.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+from compass.validate import compare_variables, compare_timers
+from compass.ocean.tests.global_ocean.forward import ForwardTestCase, \
+    ForwardStep
+
+
+class PerformanceTest(ForwardTestCase):
```
Comment (xylar) [inline compass/ocean/tests/global_ocean/lagrangian_particles/__init__.py:6]:
```suggestion
class LagrangianParticles(ForwardTestCase):
```

File: compass/ocean/tests/global_ocean/lagrangian_particles/__init__.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    """
+    A test case for performing a short forward run with an MPAS-Ocean global
+    initial condition assess performance and compare with previous results
```
Comment (xylar) [inline compass/ocean/tests/global_ocean/lagrangian_particles/__init__.py:9]: Replace this

File: compass/ocean/tests/global_ocean/init/initial_state.py (hunk 1 above)
Comment (xylar) [reply compass/ocean/tests/global_ocean/init/initial_state.py:169]: I think this belong as an optional part of the `ForwardStep`.  You should add an argument to the constructor of `ForwardStep` to indicate that particles should be included.  Then, store that argument in the class (e.g. `self.with_particles = with_particles`).  In the `run()` method, you can call this function.  That will be with the number of cores used for running the forward step, not for init (which is what would happen here, and which isn't appropriate).

**Comment by xylar** [comment]
I gather that your test case is mostly still a copy of the `PerformanceTest` at this point, right?  Sorry if I jumped the gun with some of my comments.

File: compass/ocean/tests/global_ocean/init/initial_state.py (hunk 1 above)
Comment (xylar) [reply compass/ocean/tests/global_ocean/init/initial_state.py:171]: Let me know if this is something you'd like help with.  I could make the commit to update `ForwardStep` if that would help you make progress on this PR.

**Comment by xylar** [comment]
@mark-petersen, could you close this one for now and re-open it when you're ready to make progress?

**Comment by mark-petersen** [comment]
This is no longer a priority. Closing for now.

**Pull request closed without merging**